clients/spark-extensions/pom.xml

```diff
@@ -63,7 +58,6 @@
     <dependency>
       <groupId>org.apache.iceberg</groupId>
       <artifactId>iceberg-spark3-runtime</artifactId>
-      <scope>test</scope>
```

Comment:
this has to be a test scope. The code has no dependency on iceberg runtime. This would ship the iceberg runtime w/ delta artifacts, which we don't want

Reply:
`NessieUtils` is using the `NessieClient`, therefore I removed the `test` scope so that `NessieUtils` uses whatever Nessie version that comes with `iceberg-spark3-runtime`

Reply:
yes, but if we do that then the iceberg runtime (all 100MB of it) becomes a hard dependency. We don't want that. This module isn't iceberg specific and we don't want to add iceberg as a dependency